scripts/asm-installer/vsphere/attach_vm

```diff
@@ -0,0 +1,226 @@
+#!/bin/bash
+
+set -ex
+set -o pipefail
+set +o noclobber
+set +o verbose
+
+main() {
+  echo "============================================="
+  read_args "${@}"
+  write_config_context_vm
+
+  echo "[Step 1] Mount Preparation Artifacts to VM ${VM_APP}@${VM_IP}"
+  prepare_vm
+
+  echo "[Step 2] Register VM with WorkloadGroup at namespace: ${VM_NAMESPACE}"
+  create_workloadgroup
+
+  echo "[Step 3] istioctl x workload entry and copy to vm ${VM_APP}@${VM_IP}"
+  configure_vm_transfer_file
+
+  echo "[Step 4] Install Istio Sidecar and Start Istio on VM"
+  runagent
+}
+
+prepare_vm() {
+  cd ${WORK_DIR}
+  if [[ ! -e "runagent.sh" ]]; then
+    fatal "file runagent.sh does not exists in working dir ${WORK_DIR}"
+  fi
+
+  scp -i "${KEY_FILE}" -o StrictHostKeyChecking=no runagent.sh root@${VM_IP}:"${VM_DIR}"
+  scp -i "${KEY_FILE}" -o StrictHostKeyChecking=no istio-sidecar.rpm root@${VM_IP}:"${VM_DIR}"
+}
+
+create_workloadgroup() {
+  cd ${OUTPUT_DIR}/vm-${VM_APP}
+
+  # Register VM and create a service for the VM
+  cat <<EOF >workloadgroup.yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: WorkloadGroup
+metadata:
+  name: "${VM_APP}"
+  namespace: "${VM_NAMESPACE}"
+spec:
+  metadata:
+    labels:
+      app: "${VM_APP}"
+  template:
+    serviceAccount: "${SERVICE_ACCOUNT}"
+    network: "${VM_NETWORK}"
+EOF
+
+  # add more labels for vm workloadgroup
```

Comment:
Have you tried that multiple labels for VM work? For example, add labels "channel:prod;version:1" to vm1, and labels "channel:prod" to vm2. Apply an authorization policy that select "labels: channel:prod", another authorization policy that select both "channel" and "version" labels, see if they are working correctly.